scripts/deployed_general_agent_viewer.py

```diff
@@ -117,3 +123,55 @@ def starting_balance(self) -> float:
 for session in sessions:
     with st.expander(f"{session.start_time} - {session.end_time}"):
         display_chat_history(session)
+
+st.subheader("Tool Usage")
+agent = build_agent(
+    market_type=MARKET_TYPE,
+    model="foo",  # placeholder, not used
+    system_prompt="foo",  # placeholder, not used
```

Comment:
Ah I guess this is necessary only because of agent's functions?

```
functions.extend([f(agent=agent) for f in AGENT_FUNCTIONS])
```

So if https://github.com/gnosis/prediction-market-agent/issues/253 is resolved, this dummy build_agent could be replaced with `build_agent_functions`.

@coderabbitai Please add this comment to the issue above.

Reply:
He actually did it! Super cool 🐰 

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/d3221ec7-a49b-4f35-aa66-9dd100863b4e)

Reply:
> Ah I guess this is necessary only because of agent's functions?
> ...
> So if #253 is resolved, this dummy build_agent could be replaced with `build_agent_functions`.

Yes, that's right